specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/webapplicationfirewall.json

```diff
@@ -948,6 +955,75 @@
         }
       }
     },
+    "AllowlistEntry": {
+      "required": [
+        "matchVariable",
+        "valueMatchOperator"
+      ],
+      "description": "Allows a request when the condition is satisfied.",
+      "properties": {
+        "matchVariable": {
+          "type": "string",
+          "enum": [
+            "RequestURI",
+            "RemoteAddr",
+            "RequestHeader"
+          ],
+          "description": "The variable on which we evaluate the allowlist condition",
+          "x-ms-enum": {
+            "name": "AllowlistEntryMatchVariable",
+            "modelAsString": true
+          }
+        },
+        "values": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Allowed values for the matchVariable"
+        },
+        "valueMatchOperator": {
+          "type": "string",
+          "enum": [
+            "Equals",
+            "Contains",
+            "StartsWith",
+            "EndsWith",
+            "IPMatch"
+          ],
+          "x-ms-enum": {
+            "name": "AllowlistEntryValueMatchOperator",
+            "modelAsString": true
+          },
+          "description": "Operates on the allowed values for the matchVariable"
+        },
+        "selectorMatchOperator": {
+          "type": "string",
+          "enum": [
+            "Equals",
+            "Contains",
+            "StartsWith",
+            "EndsWith"
+          ],
+          "description": "When matchVariable is a collection (e.g, headers), operate on the selector",
```

Comment:
matchVariable (above) is defined as an enum, wht do you mean its a collection ?

Reply:
MatchVariable is used to identify a request parameter. The above note is about when that parameter is a collection. For example, request header is a key-value pair i.e, it is a collection of 2 fields on which the allowlist will be defined.

Reply:
OK, so interpretation of this sample:

"matchVariable": "RequestHeader",
              "selectorMatchOperator": "StartsWith",
              "selector": "test",
              "valueMatchOperator": "Contains",
              "values": [
                "val1",
                "val2",
                "val3"
              ],

is that for a given requestHeaders dict:

{{"x-ms-fooHeader", "fooval"}, {"testHeader", "val1"}, {"test", "testValue"}}

Only {"testHeader", "val1"}   will match, as headerName.StartsWith("test")  is true AND its headerValue.Contains"val1" is true ?

Reply:
You may want to make the description more explicit for:

"selectorMatchOperator": {
           "description": "When matchVariable is a collection (e.g, headers), operate on the selector",
 "selector": {
          "description": "When matchVariable is a collection (e.g, headers), this identifies the collection's key."

Reply:
> OK, so interpretation of this sample:
> 
> "matchVariable": "RequestHeader", "selectorMatchOperator": "StartsWith", "selector": "test", "valueMatchOperator": "Contains", "values": [ "val1", "val2", "val3" ],
> 
> is that for a given requestHeaders dict:
> 
> {{"x-ms-fooHeader", "fooval"}, {"testHeader", "val1"}, {"test", "testValue"}}
> 
> Only {"testHeader", "val1"} will match, as headerName.StartsWith("test") is true AND its headerValue.Contains"val1" is true ?

Yes that's correct

Reply:
"selectorMatchOperator": {
   "description": "When the matchVariable points to a key-value pair (e.g, RequestHeader), this operates on the selector",
"selector": {
   "description": "When the matchVariable points to a key-value pair (e.g, RequestHeader), this identifies the key."

^ Is this clearer?

Reply:
> What happens if customer provides selectoreMatchOperator/selector (used for collection vars), but the matchVariable is single value variable ?

RP will throw an appropriate error in this case. All such validations will be done by RP.